### PR TITLE
remove marks and axis

### PR DESCRIPTION
**Pull request by maartenbreddels**
I noticed the Mark's `remove` method was not called, this fixes that, and I think the same for axis. As discussed with @jasongrout

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
Should we also call remove on the interaction and the scales?

**Comment by maartenbreddels** [comment]
there is also `interaction_view`, I'm not sure I know bqplot good enough to cover all of these cases.

File: js/src/Figure.js [hunk 1]
```diff
@@ -635,6 +635,12 @@ var Figure = widgets.DOMWidgetView.extend({
         this.title.text(this.model.get("title"));
     },
 
+    remove: function() {
+        Figure.__super__.remove.apply(this, arguments);
+        this.mark_views.remove()
+        this.axis_views.remove()
```
Comment (ssunkara1) [inline js/src/Figure.js]: Should they be called in the reverse order? Generally for a `destructor`, we `destory` the `child` first and then the `parent`, no?

**Comment by ssunkara1** [comment]
I thought we had this before. Don't know where it went! Thanks :)

**Comment by maartenbreddels** [comment]
@ssunkara1 will you handle any other views that need deleting?

**Comment by jasongrout** [comment]
> @ssunkara1 will you handle any other views that need deleting?

+1. If this looks good, perhaps we should merge and iterate.

File: js/src/Figure.js (hunk 1 above)
Comment (maartenbreddels) [inline js/src/Figure.js]: Yes, I'll do that and see if it works.

**Comment by ssunkara1** [comment]
@maartenbreddels Thanks. This looks good. Merging.

**Comment by maartenbreddels** [comment]
Great :+1:

**Comment by SylvainCorlay** [comment]
Thanks @maartenbreddels

**Pull request merged**